infra/src/test/kotlin/fi/fta/geoviite/infra/velho/VelhoServiceIT.kt

```diff
@@ -1,62 +1,197 @@
 package fi.fta.geoviite.infra.velho
 
+import PVAssignment
+import PVCode
+import PVDocument
+import PVDocumentStatus
+import PVId
+import com.fasterxml.jackson.databind.ObjectMapper
 import fi.fta.geoviite.infra.ITTestBase
-import org.junit.jupiter.api.AfterEach
+import fi.fta.geoviite.infra.common.Oid
+import fi.fta.geoviite.infra.util.setUser
+import fi.fta.geoviite.infra.velho.PVDictionaryType.*
 import org.junit.jupiter.api.BeforeEach
 import org.junit.jupiter.api.Test
 import org.springframework.beans.factory.annotation.Autowired
-import org.springframework.boot.autoconfigure.condition.ConditionalOnBean
+import org.springframework.beans.factory.annotation.Value
 import org.springframework.boot.test.context.SpringBootTest
 import org.springframework.test.context.ActiveProfiles
 import java.time.Instant
 import kotlin.test.assertEquals
 import kotlin.test.assertNotNull
 
+val dictionaries: Map<PVDictionaryType, List<PVDictionaryEntry>> = mapOf(
+    DOCUMENT_TYPE to listOf(
+        PVDictionaryEntry("dokumenttityyppi/dt01", "test doc type 1"),
+        PVDictionaryEntry("dokumenttityyppi/dt02", "test doc type 2"),
+    ),
+    MATERIAL_STATE to listOf(
+        PVDictionaryEntry("aineistotila/tila01", "test mat state 1"),
+        PVDictionaryEntry("aineistotila/tila02", "test mat state 2"),
+    ),
+    MATERIAL_CATEGORY to listOf(
+        PVDictionaryEntry("aineistolaji/al00", "test mat category 0"),
+        PVDictionaryEntry("aineistolaji/al01", "test mat category 1"),
+    ),
+    MATERIAL_GROUP to listOf(
+        PVDictionaryEntry("aineistoryhma/ar00", "test mat group 0"),
+        PVDictionaryEntry("aineistoryhma/ar01", "test mat group 1"),
+    ),
+    TECHNICS_FIELD to listOf(
+        PVDictionaryEntry("tekniikka-ala/ta00", "test tech field 0"),
+        PVDictionaryEntry("tekniikka-ala/ta01", "test tech field 1"),
+    )
+)
+
 @ActiveProfiles("dev", "test")
-@SpringBootTest
-@ConditionalOnBean(VelhoClient::class)
+@SpringBootTest(properties = ["geoviite.projektivelho=true"])
 class VelhoServiceIT @Autowired constructor(
-    private val fakeVelhoService: FakeVelhoService,
+    @Value("\${geoviite.projektivelho.test-port:12345}") private val velhoPort: Int,
     private val velhoService: VelhoService,
     private val velhoDao: VelhoDao,
+    private val jsonMapper: ObjectMapper,
 ) : ITTestBase() {
-    lateinit var fakeVelho: FakeVelho
 
-    @BeforeEach
-    fun startServer() {
-        fakeVelho = fakeVelhoService.start()
```

Comment:
startteja ja stoppeja ei kutsuta enää näin vaan fakeVelho on AutoClosable ja sitä käytetään tyyliin ```fakeVelho().use { fakeVelho -> // testikoodi }``` . Tuo vastaa siis Javan try-with-resources rakennetta joka hoitaa close() kutsun lopuksi, riippumatta mitä ajon aikana tapahtuu.